src/Compilers/CSharp/Portable/Symbols/Source/LocalFunctionSymbol.cs

```diff
@@ -371,6 +385,9 @@ private ImmutableArray<TypeParameterSymbol> MakeTypeParameters(DiagnosticBag dia
                         ImmutableArray.Create(location),
                         ImmutableArray.Create(parameter.GetReference()));
 
+                // Force attribute binding for diagnostics
+                typeParameter.GetAttributesBag(diagnostics);
+
```

Comment:
We are doing more and more while the symbol is being created, it looks like this function is called from the constructor. This feels dangerous, might potentially be a source of a circularity. We need a symbol to create binders, if we try to bind something at the time the symbol is created, we might try to create the binder again, and so on, and so on. We should avoid performing any binding while we create a symbol for a local function.